### PR TITLE
sql: default to SHARED service in CREATE TENANT

### PR DESCRIPTION
**Pull request by dt**
creating directly into a desired service mode avoids an extra step when setting up a new tenant, and opens up a path to creating directly in SHARED by default in the future once creations which do not wish for immediate service can opt into one of the other modes during creation.

Release note: none.
Epic: none.

### CODE REVIEW

**Review by stevendanna — APPROVED** [review]
:lgtm: 

We could have a little more in the commit message about why we think this is useful, bit I think it clearly is.

**Comment by dt** [comment]
@stevendanna I added an additional commit with a larger behavior change that I think is the real motivation. RFAL.

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_manager_test.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func TestRevertTenantToTimestamp(t *testing.T) {
 	systemSQL.Exec(t, "SET CLUSTER SETTING kv.closed_timestamp.target_duration = '100ms'")
 	systemSQL.Exec(t, "SET CLUSTER SETTING kv.closed_timestamp.side_transport_interval ='100ms'")
 	systemSQL.Exec(t, "SET CLUSTER SETTING kv.rangefeed.closed_timestamp_refresh_interval ='100ms'")
-	systemSQL.Exec(t, "CREATE VIRTUAL CLUSTER target")
+	systemSQL.Exec(t, "CREATE VIRTUAL CLUSTER target WITH SERVICE NONE")
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_manager_test.go:49]: Can we open an issue for the problem we discovered here? I think we found some pretty suspect behavior on rapid state changes.

**Review by stevendanna — APPROVED** [review]
I like the addition of this with option.

I think making SHARED the default makes the most sense. EXTERNAL certainly shouldn't be the default, and every other user will want SHARED.

As I believe you are already on top of we need to coordinate with @JeffSwenson about how this impacts his work.

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_manager_test.go (hunk 1 above)
Comment (dt) [reply pkg/ccl/streamingccl/streamingest/stream_ingestion_manager_test.go:49]: https://github.com/cockroachdb/cockroach/issues/119724

**Comment by dt** [comment]
TFTR!

bors r+

**Pull request closed without merging**